### ISSUE TITLE
100% Test Coverage

### ISSUE THREAD

**Issue by Uzlopak** [nodejs/undici#3067]
Obligatory issue. 

Also: If we have 100% test coverage, we could drop codecov.

**Comment by mertcanaltin** [comment]
Wouldn't it be good to keep the codecov for future testing?

**Comment by mertcanaltin** [comment]
> We can use c8 and configure it so that it return non 0 return value when running in the CI/CD Pipeline. So if we have a PR without a proper test, we could nag about the missing test. Codecov is only necessary if we have below 100% coverage.

I got it, thank you very much

**Comment by KhafraDev** [comment]
I don't see the benefit in having 100% code coverage.

**Comment by Uzlopak** [comment]
I am not saying, that you personally should start implementing tests.

E.g. I am still writing tests for cookies and I constantly find bugs. And I can not fix them, because upstreaming in deno is basically impossible, because I see that they cant even invest time to review the maxAge bug. I will open for this a separate issue. But yeah, it is annoying.

**Comment by KhafraDev** [comment]
100% test coverage and validating logic are mutually exclusive issues, the latter one I have no problems with. I see it as not only a false sense of security, but also a waste of time.

**Comment by Uzlopak** [comment]
I disagree. Code which is not covered by any test has imho the inherent assumption to be erroneous or being obsolete.

**Comment by KhafraDev** [comment]
I consider tests that exist for the sake of code coverage erroneous and obsolete 😉

**Comment by Uzlopak** [comment]
We agree to disagree
